src/librustc_resolve/lib.rs

```diff
@@ -1335,9 +1330,13 @@ impl<'a> Resolver<'a> {
 
     fn macro_def(&self, mut ctxt: SyntaxContext) -> DefId {
         loop {
-            match self.macro_defs.get(&ctxt.outer_expn()) {
-                Some(&def_id) => return def_id,
-                None => ctxt.remove_mark(),
+            if ctxt.outer_expn() == ExpnId::root() {
+                return DefId::local(CRATE_DEF_INDEX);
```

Comment:
This is a pretty strange case.
Could you add an explanation for it?

Reply:
This change became unecessary after moving the `DefId` into `ExpnData`.